### PR TITLE
`containerservice` - upgrade sdk to `2023-01-02-preview`

### LINKED ISSUE #20455: Rename AKS sku_tier
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Rename SKU Tier `Paid` to `Standard`. More information on https://learn.microsoft.com/en-us/azure/aks/free-standard-pricing-tiers#uptime-sla

Reference to attribute: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#sku_tier

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Potential Terraform Configuration

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by stephybun**
Closes #20455

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -1114,6 +1114,7 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice([]string{
 					string(managedclusters.ManagedClusterSKUTierFree),
 					string(managedclusters.ManagedClusterSKUTierPaid),
+					string(managedclusters.ManagedClusterSKUTierStandard),
```
Comment (tombuildsstuff) [inline internal/services/containers/kubernetes_cluster_resource.go:1117]: this'll also want documenting - or was this supposed to be part of another PR?

**Review by tombuildsstuff — APPROVED** [review]
One comment but otherwise 👍

File: internal/services/containers/kubernetes_cluster_resource.go (hunk 1 above)
Comment (stephybun) [reply internal/services/containers/kubernetes_cluster_resource.go:1117]: Nope, I forgot. Thanks!
